contracts/interfaces/IbcReceiver.sol

```diff
@@ -12,16 +12,11 @@ import {ChannelOrder, CounterParty, IbcPacket, AckPacket} from "../libs/Ibc.sol"
  * handshake callbacks.
  */
 interface IbcChannelReceiver {
-    function onOpenIbcChannel(
-        string calldata version,
-        ChannelOrder ordering,
-        bool feeEnabled,
-        string[] calldata connectionHops,
-        CounterParty calldata counterparty
-    ) external returns (string memory selectedVersion);
+    function onChannelOpenInit(string calldata version) external returns (string memory selectedVersion);
 
-    function onConnectIbcChannel(bytes32 channelId, bytes32 counterpartyChannelId, string calldata counterpartyVersion)
-        external;
+    function onChannelOpenTry(string calldata counterpartyVersion) external returns (string memory selectedVersion);
+
+    function onChannelConnect(bytes32 channelId, string calldata counterpartyVersion) external;
```

Comment:
this one did not seem like it would benefit from a split so I left the same callback for both ack and confirm steps. Open to other options though.

Reply:
will finally split as mentioned in previous comment so we are more consistent.